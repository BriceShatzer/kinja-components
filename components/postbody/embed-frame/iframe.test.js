import * as React from 'react';
import { shallow } from 'enzyme';

import { Iframe } from './';
import Dimension from 'postbody/Dimension';

const dimensions = {
	width: new Dimension(640, 'Pixel'),
	height: new Dimension(320, 'Pixel')
};

const nonSecureSource = 'http://non-whitelisted.com';
const source = 'https://bloomberg.com/foo';

describe('<Iframe />', () => {
	it('should render empty if the domain is not whitelisted & isn\'t a starter', () => {
		const props = {
			source: nonSecureSource,
			isStarter: false
		};
		const wrapper = shallow(<Iframe {...props} />);
		expect(wrapper.getElement()).toBe(null);
	});
	describe('sizes', () => {
		it('flex with dimensions', () => {
			const props = {
				aspectRatio: 'Flex',
				scrollable: true,
				source,
				isLazy: true,
				...dimensions
			};
			const wrapper = shallow(<Iframe {...props} />);
			expect(wrapper).toMatchSnapshot();
		});
		it('flex without dimensions', () => {
			const props = {
				aspectRatio: 'Flex',
				scrollable: true,
				source,
				isLazy: true
			};
			const wrapper = shallow(<Iframe {...props} />);
			expect(wrapper).toMatchSnapshot();
		});
		it('custom with dimensions', () => {
			const props = {
				aspectRatio: 'Custom',
				scrollable: true,
				source,
				isLazy: true,
				...dimensions
			};
			const wrapper = shallow(<Iframe {...props} />);
			expect(wrapper).toMatchSnapshot();
		});
		it('custom without dimensions', () => {
			const props = {
				aspectRatio: 'Custom',
				scrollable: true,
				source,
				isLazy: true
			};
			const wrapper = shallow(<Iframe {...props} />);
			expect(wrapper).toMatchSnapshot();
		});
		it('fixed with dimensions', () => {
			const props = {
				aspectRatio: 'Fixed',
				scrollable: true,
				source,
				isLazy: true,
				...dimensions
			};
			const wrapper = shallow(<Iframe {...props} />);
			expect(wrapper).toMatchSnapshot();
		});
		it('fixed without dimensions', () => {
			const props = {
				aspectRatio: 'Fixed',
				scrollable: true,
				source,
				isLazy: true
			};
			const wrapper = shallow(<Iframe {...props} />);
			expect(wrapper).toMatchSnapshot();
		});
		it('fixed with wide dimensions', () => {
			const props = {
				aspectRatio: 'Fixed',
				scrollable: true,
				source,
				isLazy: true,
				height: dimensions.height,
				width: new Dimension(1000, 'Pixel')
			};
			const wrapper = shallow(<Iframe {...props} />);
			expect(wrapper).toMatchSnapshot();
		});
	});
	it('non-lazy should use src instead of data-src', () => {
		const props = {
			aspectRatio: 'Flex',
			scrollable: true,
			source
		};
		const wrapper = shallow(<Iframe {...props} />);
		expect(wrapper.find('iframe').prop('src')).toBe(source);
		expect(wrapper.find('iframe').prop('data-src')).not.toBeDefined();
	});
	it('whitelisted url should not be sandboxed', () => {
		const props = {
			aspectRatio: 'Flex',
			scrollable: true,
			source
		};
		const wrapper = shallow(<Iframe {...props} />);
		expect(wrapper.find('iframe').prop('sandbox')).toBe(undefined);
	});
	it('should be able to be non-scrollable', () => {
		const props = {
			aspectRatio: 'Flex',
			scrollable: false,
			source
		};
		const wrapper = shallow(<Iframe {...props} />);
		expect(wrapper.find('iframe').prop('scrolling')).toBe('no');
	});
	describe('SSL placeholder', () => {
		it('should render an SSL placeholder', () => {
			const props = {
				source: nonSecureSource
			};
			const wrapper = shallow(<Iframe {...props} />);
			expect(wrapper).toMatchSnapshot();
		});
		it('fixed aspect ratio', () => {
			const props = {
				source: nonSecureSource,
				aspectRatio: 'Fixed'
			};
			const wrapper = shallow(<Iframe {...props} />);
			expect(wrapper).toMatchSnapshot();
		});
		it('wide dimension', () => {
			const props = {
				source: nonSecureSource,
				width: new Dimension(1000, 'Pixel')
			};
			const wrapper = shallow(<Iframe {...props} />);
			expect(wrapper).toMatchSnapshot();
		});
		it('with thumbnail', () => {
			const props = {
				source: nonSecureSource,
				thumbnail: {
					id: 'id',
					format: 'jpg'
				}
			};
			const wrapper = shallow(<Iframe {...props} />);
			expect(wrapper).toMatchSnapshot();
		});
	});
});
