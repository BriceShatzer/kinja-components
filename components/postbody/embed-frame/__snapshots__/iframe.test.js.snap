// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Iframe /> SSL placeholder fixed aspect ratio 1`] = `
<WithPlatform
  aspectRatio="Fixed"
  domain="non-whitelisted.com"
  source="http://non-whitelisted.com"
/>
`;

exports[`<Iframe /> SSL placeholder should render an SSL placeholder 1`] = `
<WithPlatform
  domain="non-whitelisted.com"
  source="http://non-whitelisted.com"
/>
`;

exports[`<Iframe /> SSL placeholder wide dimension 1`] = `
<WithPlatform
  domain="non-whitelisted.com"
  source="http://non-whitelisted.com"
  width={
    Dimension {
      "unit": "Pixel",
      "value": 1000,
    }
  }
/>
`;

exports[`<Iframe /> SSL placeholder with thumbnail 1`] = `
<WithPlatform
  domain="non-whitelisted.com"
  source="http://non-whitelisted.com"
  thumbnail={
    Object {
      "format": "jpg",
      "id": "id",
    }
  }
/>
`;

exports[`<Iframe /> sizes custom with dimensions 1`] = `
<p
  className=""
>
  <iframe
    className="lazyload custom"
    data-src="https://bloomberg.com/foo"
    height="320px"
    scrolling="yes"
    width="640px"
  />
</p>
`;

exports[`<Iframe /> sizes custom without dimensions 1`] = `
<p
  className=""
>
  <iframe
    className="lazyload custom"
    data-src="https://bloomberg.com/foo"
    scrolling="yes"
  />
</p>
`;

exports[`<Iframe /> sizes fixed with dimensions 1`] = `
<p
  className="has-media has-embed"
>
  <iframe
    className="lazyload custom aspect"
    data-src="https://bloomberg.com/foo"
    height="320px"
    scrolling="yes"
    width="640px"
  />
</p>
`;

exports[`<Iframe /> sizes fixed with wide dimensions 1`] = `
<p
  className="has-media has-embed has-video media-large"
>
  <iframe
    className="lazyload custom aspect"
    data-src="https://bloomberg.com/foo"
    height="320px"
    scrolling="yes"
    width="1000px"
  />
</p>
`;

exports[`<Iframe /> sizes fixed without dimensions 1`] = `
<p
  className="has-media has-embed"
>
  <iframe
    className="lazyload custom aspect"
    data-src="https://bloomberg.com/foo"
    scrolling="yes"
  />
</p>
`;

exports[`<Iframe /> sizes flex with dimensions 1`] = `
<p
  className=""
>
  <span
    className="clear-both flex-video video-320 widescreen"
  >
    <iframe
      allowFullScreen="allowfullscreen"
      className="lazyload"
      data-src="https://bloomberg.com/foo"
      height="320px"
      scrolling="yes"
      width="640px"
    />
  </span>
</p>
`;

exports[`<Iframe /> sizes flex without dimensions 1`] = `
<p
  className=""
>
  <span
    className="clear-both flex-video video-unknown widescreen"
  >
    <iframe
      allowFullScreen="allowfullscreen"
      className="lazyload"
      data-src="https://bloomberg.com/foo"
      scrolling="yes"
    />
  </span>
</p>
`;
